### DISCUSSION TITLE
unable to access the functions.php

### DISCUSSION THREAD

**Discussion by rajpatel2435** [flyntwp/flynt#555] [Q&A] [answered]
Hi Guys!!

Hope you are doing welllllll.

One question related to use of some function inside functions.php inside flynt.

Unable to access defined function insider functions.php 
![image](https://github.com/flyntwp/flynt/assets/81604941/0d49eef9-222f-4fe1-aa95-1827167721cd)

I want to use function here 
![image](https://github.com/flyntwp/flynt/assets/81604941/5fc85de5-59e2-495f-a97e-45cbb85d9e9d)

it says function is not defined.

Can anyone help me with that please?

Thanks

**Comment by timohubois** [comment]
@rajpatel2435 at the moment I don't understand what you are trying to achieve, but from the screenshots I see that you are using .php files within the templates folder and I would strongly recommend not doing this.

Can you please describe in more detail what exactly you want to do?

**Comment by rajpatel2435** [comment]
Hey @timohubois  @aaronmeder I want to use the WordPress template functionality where I have template file in php format and I have some predefined functions in php file.
So in WordPress whatever we type in functions.php file we can access that in all over the template for example If i defined the function in root panel of functions.php I can acees that functions in template.php file So I dont need to write function again and again.

what is the template  hierarchy for Flynt. what if we want use defined function with php? Do we need  component for that?

**Comment by aaronmeder** [comment]
@rajpatel2435 it's not fully understandable to me what your goal would be.
 
A big part of what makes Flynt is the included [Timber](https://timber.github.io/docs/v2/), which comes with [Twig](https://twig.symfony.com/) as included templating language. So I think in most cases you would not want to use the traditional php template files (nor `get_template_part()`) but use twig for creating your templates (basic example is single.php and then that loads a twig file).

If what you want however is to define some global helper functions that can be used everywhere, for example in your Component's functions.php then have a look at the `lib/Utils/StringHelpers.php` file as an example:
It provides a helper function called `camelCaseToKebab` to convert a string into a different format. To use this function, as an example, in `Components/BlockWysiwyg/functions.php` you would add this code:

```php
use Flynt\Utils\StringHelpers;
add_filter('Flynt/addComponentData?name=BlockWysiwyg', function ($data, $componentName) {
    $data['someText'] = StringHelpers::camelCaseToKebab('howToTurnACamelIntoASnake');

    return $data;
}, 10, 2);
```

In that Component's twig template (Components/BlockWysiwyg/index.twig) you can output that using `{{ someText }}`.

Did that answer some of your questions?

Happy weekend!
Aaron

**Comment by rajpatel2435** [comment]
Hi @aaronmeder @timohubois,

Got that!

Thank you so much for your response.

As Flynt provides a speed score of 100, I am building my WordPress theme with the help of Flynt. However, I am currently experiencing a performance issue where the mobile score is around 57, primarily due to Total Blocking Time (TBT).

Could you assist me in improving this? I would appreciate any insights into where I might be going wrong.

Here is the site: [Baltimore Ravens](https://wordpress-1233981-4407119.cloudwaysapps.com/nfl/teams/baltimore-ravens/)

![image](https://github.com/flyntwp/flynt/assets/81604941/1194c55e-0443-444e-a5ac-f0619a0ade4a)


Thanks,

**Reply by timohubois** [reply]
@rajpatel2435 If you look at the website's network, you will find assets (e.g. Javascript and CSS) that are loaded by external services. Can you run the test without loading them and will the results be the same? From what I've seen so far, I would prefer a different strategy for loading assets from external servers, but there are several options and the type may differ depending on specific requirements.

But that is a completely different topic. And I don't think it should be discussed in this thread. 

If you have solved your problem, it would be nice if you could briefly write down what you did here to help others who may have the same question. After that, it would be great if you could close this discussion.

For further questions you can always start a new discussion.

Best,
Timo

**Comment by rajpatel2435** [comment, marked as answer]
Hi @timohubois @aaronmeder 

Thank you for your answer.

Regarding that, I enabled the component based structure in order to load the template with rewrite rule.

Thanks
